### PR TITLE
SCT-1147 Start on storage API, update design

### PR DESCRIPTION
**Pull request by MrCreosote**
- removed metadata on mappings from design - could be 100B mappings. Use
descriptive namespace names instead (e.g. KBase Genome Objects, KBase
Assembly Objects)

-removed some pointless returns from the API

### CODE REVIEW

File: src/jgikbase/idmapping/core/HashedToken.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'''
```
Comment (jayrbolton) [inline src/jgikbase/idmapping/core/HashedToken.py]: Style nitpicks:
- You are using more java-styled directory structure. In python, a directory is typically a package containing modules. So it doesn't make sense to have packages named `src`, `jgikbase`, or `core`. Typically there is just one directory called `idmapping` (name of your actual package). See here: http://docs.python-guide.org/en/latest/writing/structure/
- You might consider naming your modules lowercase with underscores, as per pep: https://www.python.org/dev/peps/pep-0008/#package-and-module-names
- Pep also calls for double-quote docstrings: https://www.python.org/dev/peps/pep-0257/
- In python 3 code, I always see classes as `class MyClass:` without the extra `(object)` which isn't necessary

**Comment by jayrbolton** [comment]
Added one style nitpick comment. Will have to read more about the design to comment on the other classes you made.

**Comment by MrCreosote** [comment]
> You are using more java-styled directory structure. In python, a directory is typically a package containing modules. 

Yeah, I've never really understood that. It's begging for name collisions (see magic)

> You might consider naming your modules lowercase with underscores,

There's a TODO in the PR for that. I'll do it in another PR. Brain fart.

> Pep also calls for double-quote docstrings

That's PyDev's fault. I'll fix in another PR.

> In python 3 code, I always see classes as class MyClass:

I'll fix in another PR

**Comment by MrCreosote** [comment]
src is necessary because otherwise pydev analyses everything in the entire repo.

**Comment by MrCreosote** [comment]
I've fixed the style stuff, although I may want to shuffle around the packages and modules a bit, but I'll do that as I get to it. It's 4 PRs down the road at this point.

**Pull request merged**